storage/gcloud/aio/storage/blob.py

```diff
@@ -73,23 +75,33 @@ async def get_signed_url(  # pylint: disable=too-many-locals
             raise ValueError("expiration time can't be longer than 604800 "
                              'seconds (7 days)')
 
-        iam_client = iam_client or IamClient(service_file=service_file,
-                                             token=token, session=session)
-
-        quoted_name = quote(self.name, safe='')
-        canonical_uri = f'/{self.bucket.name}/{quoted_name}'
+        quoted_name = quote(self.name, safe=b'/~')
+        canonical_uri = f'/{quoted_name}'
 
         datetime_now = datetime.datetime.utcnow()
         request_timestamp = datetime_now.strftime('%Y%m%dT%H%M%SZ')
         datestamp = datetime_now.strftime('%Y%m%d')
 
-        service_account_email = (service_account_email
-                                 or iam_client.service_account_email)
+        # TODO: support for GCE_METADATA and AUTHORIZED_USER tokens. It should
+        # be possible via API, but seems to not be working for us in some
+        # cases.
+        #
```

Comment:
How worthwhile would it be to reach out to GCP support?

Reply:
I've opened the dialog with our TAM and will improve this function over time as per their response :+1: